src/enrichments/gutter.ts

```diff
@@ -0,0 +1,122 @@
+import { rangeToTarget, IRangeTarget } from "../ranges";
+
+export interface IGutterEnrichmentProvider {
+  /** If the provider wants to add an enrichment for this target, return a button element.
+   * @param target
+   */
+  getButton(target: IRangeTarget): HTMLButtonElement | null;
+
+  /**
+   * The user clicked the provider's button, add an enrichment for the provided target.
+   * @param target
+   */
+  addEnrichment(target: IRangeTarget): void;
+}
+
+/**
+ * This manager provides support for creating gutter item enrichments from selected text in the document body.
+ * Providers register themselves with the manager and will be called when a new range is selected.
+ *
+ * The manager must be created on an element that has an `la-gutter` and an `la-akoma-ntoso` element as descendants.
+ */
+export class GutterEnrichmentManager {
+  protected root: Element;
+  protected gutter: Element | null;
+  protected akn: Element | null;
+  protected providers: IGutterEnrichmentProvider[];
+  protected floaterTimeout: number | null;
+  protected target: IRangeTarget | null;
+  protected floatingContainer: HTMLElement;
+
+  constructor (root: Element) {
+    this.root = root;
+    this.gutter = root.querySelector('la-gutter');
+    this.akn = root.querySelector('la-akoma-ntoso');
+    this.providers = [];
+    this.floatingContainer = this.createFloatingContainer();
+    this.floaterTimeout = null;
+    this.target = null;
+    document.addEventListener('selectionchange', this.selectionChanged.bind(this));
+  }
+
+  addProvider (provider: IGutterEnrichmentProvider) {
+    this.providers.push(provider);
+  }
+
+  createFloatingContainer () {
+    const item = document.createElement('la-gutter-item');
+    const btnGroup = document.createElement('div');
+    btnGroup.className = 'gutter-enrichment-new-buttons btn-group-vertical btn-group-sm bg-white';
```

Comment:
I guess we can assume that `indigo-akn` will always be used in bootstrap-dependent projects 🤔  and if not we can always style it accordingly

Reply:
Ja, that's the assumption I'm making. Otherwise we have to style it explicitly here, and we don't have a mechanism for injecting CSS here. :(